File: WebUI/ClientApp/src/domain/day.ts
import { fromShortDate, getDayName } from "../utils/date-utils";

class Day {
	constructor(
		public shortDate: string = "",
		public isHoliday: boolean = false) {
	}

	public get dayName() {
		return getDayName(this.shortDate)
	}

	public toMoment() {
		return fromShortDate(this.shortDate);
	}
}

export default Day;

File: FoodOrder.Domain/Entities/Day.cs
using System;

namespace FoodOrder.Domain.Entities {
	public class Day {
		public DateTime Date { get; set; }
		public bool IsHoliday { get; set; }

		public static Day Create(string shortDate, bool isHoliday = false) {
			return new Day {
				Date = Convert.ToDateTime(shortDate),
				IsHoliday = isHoliday,
			};
		}
	}
}